guide/java/defining-and-deploying.md

```diff
@@ -5,121 +5,171 @@ layout: website-normal
-
-        StartableApplication app = new WebClusterDatabaseExample()
-                .appDisplayName("Brooklyn WebApp Cluster with Database example")
-                .manage(server.getManagementContext());
-        
-        app.start(ImmutableList.of(loc));
-        
-        Entities.dumpInfo(app);
-    }
+{% readj gist_generator/GistGeneratorYamlTest.java %}
 {% endhighlight %}
 
-Compile and run this with the [``brooklyn-all`` jar]({{site.path.guide}}/start/download.html) on the classpath,
-pointing at your favourite WAR on your filesystem. 
-(If the ``import`` packages aren't picked up correctly,
-you can cheat by looking at [the file in Github](https://github.com/apache/brooklyn-library/blob/master/examples/simple-web-cluster/src/main/java/org/apache/brooklyn/demo/WebClusterDatabaseExample.java);
-and you'll find a sample WAR which uses the database as configured above 
-[here](http://search.maven.org/remotecontent?filepath=org/apache/brooklyn/example/brooklyn-example-hello-world-sql-webapp/0.8.0-incubating/brooklyn-example-hello-world-sql-webapp-0.8.0-incubating.war).)
- TODO example webapp url 
- 
-If you want to adventure beyond ``localhost`` (the default),
-simply supply the your favourite cloud (e.g. ``aws-ec2:eu-west-1``)
-with credentials set up as described [here]({{ site.path.guide }}/use/guide/management/index.html#startup-config).
-
--->
+
+## Building the OSGi Bundle
+
+We will build this as an [OSGi Bundle](https://www.osgi.org/developer/architecture/) so that it
+can be added to the Apache Brooklyn server at runtime, and so multiple versions of the blueprint 
+can be managed.
+
+The `mvn clean install` will automatically do this, creating a jar inside the `target/` sub-directory
+of the project. This works by using the 
+[Maven Bundle Plugin](http://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html)
+which we get automatically by declaring the `pom.xml`'s parent as `brooklyn-downstream-parent`.
+
+
+## Adding to the catalog
+
+Similar to the `sample.bom` entity that ships with the archetype, we will define a `.bom` file
+to add our `GistGenerator` to the catalog. Substitute the URL below for your own newly built 
+artifact (which will be in the `target` sub-directory after running `mvn clean install`).
+
+{% highlight yaml %}
+{% readj gist_generator/gist_generator.bom %}
+{% endhighlight %}
+
+*Unfortunately the file org.eclipse.egit.github.core-2.1.5.jar (available on maven central) was 
+generated incorrectly as an OSGi bundle (there is a missing quotation mark from the manfest file,
+making it invalid). The above YAML references a corrected version of this OSGi bundle, made 
+available for test purposes.*
+
+The command below will use the REST api to add this to the catalog of a running Brooklyn instance.
+Substitute the credentials, URL and port for those of your server.
+
+    curl -u admin:pa55w0rd https://127.0.0.1:8443/v1/catalog --data-binary @gist_generator.bom
+
```

Comment:
In other places we've been marking bash using {% highlight bash %} and prefixing the lines with $